corehq/apps/locations/permissions.py

```diff
@@ -319,6 +319,19 @@ def user_can_access_any_location_id(domain, user, location_ids):
             .exists())
 
 
+def user_can_access_other_user(domain, user, other_user):
+    if user.has_permission(domain, 'access_all_locations'):
+        return True
+
+    accessible_location_ids = (SQLLocation.objects .accessible_to_user(domain, user) .values_list('id', flat=True))
+
+    other_user_locations = other_user.sql_locations
+    return any(map(
+        lambda location_id: location_id in accessible_location_ids,
+        other_user_locations.values_list('id', flat=True) if other_user_locations else []
```

Comment:
I think you might be able to do
`other_user_locations.filter(id__in=accessible_location_ids).count() > 0` to not have to loop through each one